### PR TITLE
cartesian_pathサンプルの追加

### PR DESCRIPTION
**Pull request by KuraZuzu**
# What does this implement/fix?

cartesian_pathサンプルを追加します。
動作概要は、[CRANE X7](https://github.com/rt-net/crane_x7_ros)のcartesian_pathサンプルと同様です。

# Does this close any currently open issues?

しません。

# How has this been tested?

Gzebo上でテストしました。

- Ubuntu 24.04 Desktop
    - 6.11.0-29-generic
- ROS 2 Jazzy
- GAZEBO Harmonic v 8.9.0

```bash
ros2 launch sciurus17_gazebo sciurus17_with_table.launch.py 
```

```bash
ros2 launch sciurus17_examples example.launch.py example:='cartesian_path' use_sim_time:='true'
```

# Any other comments?

# Checklists

- [x]  I have read the CONTRIBUTING guidelines.
    
    - If there is no guideline in the repository, [rt-net organization's guideline](https://github.com/rt-net/.github/blob/master/CONTRIBUTING.md) applies.
- [x]  I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same change.

### CODE REVIEW

File: sciurus17_examples/README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
     - [Gazeboでサンプルプログラムを実行する場合](#gazeboでサンプルプログラムを実行する場合)
   - [Examples](#examples)
     - [gripper\_control](#gripper_control)
+    - [cartesian\_path](#cartesian_path)
```
Comment (Kuwamai) [inline sciurus17_examples/README.md:14]: examplesの目次にも追加お願いします。

File: sciurus17_examples/README.md [hunk 2]
```diff
@@ -131,6 +132,21 @@ ros2 launch sciurus17_examples example.launch.py example:='gripper_control'
 
 ---
 
+### cartesian_path
+
+[Cartesian Path](https://moveit.picknik.ai/humble/doc/examples/move_group_interface/move_group_interface_tutorial.html#cartesian-paths)
```
Comment (Kuwamai) [inline sciurus17_examples/README.md]: 参照しているドキュメントがhumble版のままだったので最新のものに変更したいです。
```suggestion
[Cartesian Path](https://moveit.picknik.ai/main/doc/examples/move_group_interface/move_group_interface_tutorial.html#cartesian-paths)
```

File: sciurus17_examples/src/cartesian_path.cpp [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  // SRDFに定義されている"l_arm_init_pose"の姿勢にする
+  move_group_arm.setNamedTarget("l_arm_init_pose");
+  move_group_arm.move();
+
+  // ハンドを開く
+  gripper_joint_values[0] = angles::from_degrees(-40);
+  move_group_gripper.setJointValueTarget(gripper_joint_values);
+  move_group_gripper.move();
+
+  // 座標(x=0.3, y=0.0161 z=0.1)を中心に、XY平面上に半径0.1 mの円を3回描くように手先を動かす
```
Comment (Kuwamai) [inline sciurus17_examples/src/cartesian_path.cpp]: 中心のy座標がかなり細かい数値になっていますがこれにはなにか理由があるのでしょうか?
可動範囲の問題であれば0.1等キリのいい数値でいいかと思います。

**Review by Kuwamai — CHANGES_REQUESTED** [review]
軽微な修正点についていくつかコメントしました :bow:

File: sciurus17_examples/README.md (hunk 1 above)
Comment (KuraZuzu) [reply sciurus17_examples/README.md:14]: 追加しました。

File: sciurus17_examples/src/cartesian_path.cpp (hunk 3 above)
Comment (KuraZuzu) [reply sciurus17_examples/src/cartesian_path.cpp]: 0.161 -> 0.2 に変更しました。円を描く上で、テーブルの上のキューブにぶつからないキリの良い数値であるため0.2としました。

`0.161`とした理由は、動作前のポーズである`l_arm_init_pose`から大きく肩を動かさないような値に設定したためです。そのため、`l_link2`のY座標0.1605から四捨五入した0.161としていました。

image

**Comment by KuraZuzu** [comment]
@Kuwamai ros2ブランチの最新のコミットまで取り込んでコンフリクトを解消し、Gazeboでの動作まで確認しました。ご確認をお願いします。

File: sciurus17_examples/README.md [hunk 4]
```diff
@@ -102,6 +103,7 @@ ros2 launch sciurus17_examples example.launch.py example:='gripper_control'
 `demo.launch`を実行している状態で各サンプルを実行できます。
 
 - [gripper\_control](#gripper_control)
+- [cartesian\_path](#cartesian_path)
```
Comment (Kuwamai) [inline sciurus17_examples/README.md:108]: ドキュメントでの紹介順序はX7と同様にpose_groupstateの後でお願いします

**Review by Kuwamai — CHANGES_REQUESTED** [review]
ドキュメントについて1点コメントしました。
動作確認は来週行います。

File: sciurus17_examples/src/cartesian_path.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  // 座標(x=0.3, y=0.2 z=0.1)を中心に、XY平面上に半径0.1 mの円を3回描くように手先を動かす
+  std::vector<geometry_msgs::msg::Pose> waypoints;
+  float num_of_waypoints = 30;
+  int repeat = 3;
+  float radius = 0.1;
+
+  geometry_msgs::msg::Point center_position;
+  center_position.x = 0.3;
+  center_position.y = 0.2;
+  center_position.z = 0.1;
```
Comment (Kuwamai) [inline sciurus17_examples/src/cartesian_path.cpp]: 手が机のものに当たりやすいのでもう少し上のほうがよさそうです。
```suggestion
  center_position.z = 0.15;
```

File: sciurus17_examples/src/cartesian_path.cpp [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  auto move_group_gripper_node = rclcpp::Node::make_shared("move_group_gripper_node", node_options);
+  // For current state monitor
+  rclcpp::executors::SingleThreadedExecutor executor;
+  executor.add_node(move_group_arm_node);
+  executor.add_node(move_group_gripper_node);
+  std::thread([&executor]() {executor.spin();}).detach();
+
+  MoveGroupInterface move_group_arm(move_group_arm_node, "l_arm_group");
+  move_group_arm.setMaxVelocityScalingFactor(0.1);      // Set 0.0 ~ 1.0
+  move_group_arm.setMaxAccelerationScalingFactor(1.0);  // Set 0.0 ~ 1.0
```
Comment (Kuwamai) [inline sciurus17_examples/src/cartesian_path.cpp:49]: 他のサンプルと合わせて加速度も遅めでお願いします。
```suggestion
  move_group_arm.setMaxAccelerationScalingFactor(0.1);  // Set 0.0 ~ 1.0
```

**Review by Kuwamai — CHANGES_REQUESTED** [review]
実機の挙動を確認してコメント2点追加しました。

File: sciurus17_examples/README.md (hunk 4 above)
Comment (KuraZuzu) [reply sciurus17_examples/README.md:108]: ドキュメントの順序を変更しました。

**Comment by KuraZuzu** [comment]
レビューコメントに沿って修正しました。

File: sciurus17_examples/README.md [hunk 7]
```diff
@@ -141,10 +143,20 @@ SRDFファイル[sciurus17_moveit_config/config/sciurus17.srdf](../sciurus17_mov
 次のコマンドを実行します。
 
 ```sh
+ros2 launch sciurus17_examples example.launch.py example:='cartesian_path'
```
Comment (Kuwamai) [inline sciurus17_examples/README.md]: コマンドの追記箇所がずれています。

File: sciurus17_examples/src/cartesian_path.cpp [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  auto move_group_gripper_node = rclcpp::Node::make_shared("move_group_gripper_node", node_options);
+  // For current state monitor
+  rclcpp::executors::SingleThreadedExecutor executor;
+  executor.add_node(move_group_arm_node);
+  executor.add_node(move_group_gripper_node);
+  std::thread([&executor]() {executor.spin();}).detach();
+
+  MoveGroupInterface move_group_arm(move_group_arm_node, "l_arm_group");
+  move_group_arm.setMaxVelocityScalingFactor(0.1);      // Set 0.0 ~ 1.0
+  move_group_arm.setMaxAccelerationScalingFactor(0.1);  // Set 0.0 ~ 1.0
```
Comment (Kuwamai) [inline sciurus17_examples/src/cartesian_path.cpp]: すみません、実機で動作確認したら加減速を繰り返して動きがぎこちなくなってしまったので元に戻しておいてください 🙇 
```suggestion
  move_group_arm.setMaxAccelerationScalingFactor(1.0);  // Set 0.0 ~ 1.0
```

**Review by Kuwamai — CHANGES_REQUESTED** [review]
少しコメントしました。

File: sciurus17_examples/README.md (hunk 7 above)
Comment (KuraZuzu) [reply sciurus17_examples/README.md]: 修正しました。

**Comment by KuraZuzu** [comment]
再度修正しました。
全体のREADMEの誤字脱字や表記ゆれも併せて修正しています。

**Review by Kuwamai — APPROVED** [review]
LGTMです
